### DISCUSSION TITLE
FirstStepRuby :revisited

### DISCUSSION THREAD

**Discussion by kakutani** [rubima/magazine.rubyist.net#379] [要望・アイデア]
るびまの「常設記事」である(あった) FirstStepRubyをアップデートしたい/する/していくためのスレ。

https://magazine.rubyist.net/articles/first_step_ruby/FirstStepRuby.html

* 記事の位置づけ: インストールやチュートリアルのような説明に留まらない「Rubyの学び方」の始め方
* @takkanm 「スナップショットで新規に書き上げるより、メンテが継続できるかの方が課題」
* @ko1「いい記事ができれば人に勧めやすいし、できなければ今のように忘れられる...」

ruby-jpの #rubima で話題に上がったのですが、あちらはログが流れていってしまうのでスレにしてみました。

**Comment by hsbt** [comment]
最近 railsgirls に Ruby をプログラミングとして学ぶ最初のチュートリアルが追加されていました。なかなかよさそう。

https://guides.railsgirls.com/ruby-intro

https://www.python.org/about/gettingstarted/ Python のは今とるびまの物とあまり変わらない気がします。これ使われているのかなあ。

**Comment by snoozer05** [comment]
https://github.com/rubima/magazine.rubyist.net/issues/352#issuecomment-1443096673 ここで島田が問題提起していることとも絡んでくるんじゃないかなと感じています。

* Rubyの記事が少ないが書く場がないという状況の解決がるびまのモチベーションであった
* Rubyが普及していくに際して、（混乱のない）良いRubyの始め方の記事が欲しかった

というのが、この記事が常設されている理由なのかなと感じています（違っていたらツッコミ欲しいです！）
ですが、現在はいろんなところに記事がある状況で（例えばrailsgirlsのチュートリアルなんかもそうですし、https://igaigarb.booth.pm/items/3089926 (→もとは[るびまの記事](https://magazine.rubyist.net/articles/0059/0059-Ruby-Rails-Beginners-Guide.html)）なんかもある、るびまにこの記事があると良い理由というのはどんなところがあるのかな、というのを、まず確認しておく必要があるんじゃないかなあと。

他の記事ではこういった点はどうしてもカバーできない（るびまにしか拾えない）というのが見つけられれば、そこに絞ってアップデートできるんじゃないかなと感じています。どうでしょうね

**Reply by snoozer05** [reply]
https://www.slideshare.net/takahashim/how-to-become-a-rubyist これの記事化というのは一つ方向性としてアリだったりしないかな

**Comment by ko1** [comment]
感想です。書かないのに口だけ出すやつ。

1. るびまで Ruby に興味をもった人（いるの？）が、近くにリンクがあるとクリックしやすい場所として
2. るびま関係なく「最初はこれ読んどくといいよ」という定番として、人に勧めやすいリソースとして

という価値かなぁ、と思っています。20年位前（！！）は 1 も結構いたかもしれないけど、今はいないかもですね（reboot 次第）。
2 という点では、すでに優れた入門があるなら、そこへのリンクで全然かまわないと思います。
ただ、読みやすい形でない（高橋さんの slideshare の資料は良さそうだけど、少なくとも広告でまともに見れない）とかなら、読みやすい形で置いとくと良さそうですね。

たとえば、tryruby への導線をもう少し増やしても良さそう。

るびまじゃなくてもよい/るびまでないほうが良い、というのは、それはそれでその通りかと思います。例えば、www.ruby-lang.org にあるほうが良いかも。特にるびまでやるべき、みたいな意見はありません（るびまはそこへのリンクがあると良いと思いました）。日本語リソースへの導線が多ければ、www じゃなくてるびまで、って判断はあるかも？

「Ruby の世界ってのはこんなふうに広がってるんやで」というのがわかると良さそうな感じですかね。ざっと思いつきで書いてみると（内容は適切なページへのリンクでいいと思います）：

* 技術面
  *  ちょい試す tryruby、ほかにもクラウド環境とかある？
  * 自分のマシンにインストール
  * 簡単なチュートリアル
  * 難しいチュートリアル、公式ドキュメント、公式ドキュメントの使い方
  * Rails のチュートリアル
  * Ruby の活用事例（mruby とかもこの辺？）
* 参考書籍（2023年のおすすめ書籍は良さそう）
* コミュニティ
  * コミュニケーション手段（以前は ML だった）
  * コミュニティイベント

とかわかると雰囲気がわかるのかなぁ。あと、歴史ものとして、高橋さんの30周年の発表とか、そもそもパンフレット https://30.ruby.or.jp/assets/pdf/Ruby30.pdf 紹介してもいいかも？（せっかく作ったので紹介したい）

**Reply by ko1** [reply]
書いてもいいんだけど、もうどっぷり ruby 界隈に漬かってしまって、何が first step する人に必要とされるか選別する自信がないんだよな

**Reply by ko1** [reply]
たとえば最初にバーンと ruby-jp の #newbe でわかんなかったら気楽に聞いてね！ って書いちゃってもいかも。
